dns-controller/cmd/dns-controller/main.go

```diff
@@ -88,7 +88,13 @@ func main() {
 		os.Exit(1)
 	}
 
-	dnsController, err := dns.NewDNSController(dnsProvider, zoneRules)
+	dnsCache, err := dns.NewDNSCache(dnsProvider)
+	if err != nil {
+		glog.Errorf("Error initializing DNS cache: %v", err)
+		os.Exit(1)
+	}
+
+	dnsController, err := dns.NewDNSController(dnsCache, zoneRules)
```

Comment:
Much better ... simpler code, and avoids the public constructor for a non-public class.